apps/arena/lib/arena/game_updater.ex

```diff
@@ -1210,6 +1230,41 @@ defmodule Arena.GameUpdater do
     end
   end
 
+  defp update_visible_players(%{players: players, bushes: bushes} = game_state, game_config) do
+    Enum.reduce(players, game_state, fn {player_id, player}, game_state ->
+      bush_collisions =
+        Enum.filter(player.collides_with, fn collided_id ->
+          Map.has_key?(bushes, collided_id)
+        end)
+
+      visible_players =
+        Enum.reduce(players, [], fn {candicandidate_player_id, candidate_player}, acc ->
+          candidate_bush_collisions =
+            Enum.filter(candidate_player.collides_with, fn collided_id ->
+              Map.has_key?(bushes, collided_id)
+            end)
+
+          players_in_same_bush? =
+            Enum.any?(bush_collisions, fn collided_id -> collided_id in candidate_bush_collisions end)
+
+          players_close_enough? =
+            Physics.distance_between_entities(player, candidate_player) <= game_config.bushes.field_of_view_inside_bush
+
+          if candidate_player.id != player_id and
+               (Enum.empty?(candidate_bush_collisions) or (players_in_same_bush? and players_close_enough?)) do
+            acc ++ [candicandidate_player_id]
```

Comment:
I think there's no reason not to use `[candidate_player_id | acc]` here, right?

It's the better option out of the two, [as per ++ docs](https://hexdocs.pm/elixir/1.12/Kernel.html#++/2), and I don't think we care about what order these players are on

Reply:
nope, not reason at all. Changed!

Also i noticed we're doing extra checks since we're running the checks for the same player, i moved the ifs a little bit so we dont do extra calculations